3rdparty/openssl/append-deprecations

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+# Copyright (c) Open Enclave SDK contributors.
+# Licensed under the MIT License.
+
+##==============================================================================
+##
+## append-deprecations:
+##
+##     Append inclusion of the deprecations header to every header file under
+##     the given directory.
+##
+##==============================================================================
+
+## Check arguments.
+if [ "$#" != "1" ]; then
+    echo "Usage: $0 directory"
+    exit 1
+fi
+
+## Name arguments.
+directory=$1
+
+## If the directory does not exist:
+if [ ! -d "${directory}" ]; then
+    echo "$0: no such directory: ${directory}"
+    exit 1
+fi
+
+headers=`find ${directory} -name '*.h'`
+
+## Append the deprecation inclusion */
+for i in ${headers}
+do
+    if [[ "${i}" != *"ssl.h" ]] && [[ "${i}" != *"x509_vfy.h" ]]; then
+        continue
+    fi
+    echo "#include <openssl/deprecations.h>" >> ${i}
```

Comment:
Instead of modifying each file, an alternative is to add `-include openssl/deprecations.h` as a command line to the compiler via cmake/pkgconfig.